settings.h

```diff
@@ -1064,6 +1071,11 @@ bool settings_read_coord_data(coord_system_id_t id, float (*coord_data)[N_AXIS])
 // Temporarily override acceleration, if 0 restore to configured setting value
 bool settings_override_acceleration (uint8_t axis, float acceleration);
 
+#if ENABLE_ACCELERATION_PROFILES
+extern uint8_t ActiveAccelProfile;
```

Comment:
IMO ActiveAccelProfile should be added to gc_state and passed to the planner via pl_block instead. Set the default in gc_init() and restore it in the M2/M30 handling code.

Reply:
Would something like this work? 
this in gc_init() to initialize the state:
![gc_init](https://github.com/user-attachments/assets/b9d929ee-50a1-4ad6-a747-bfba4beff4a2)
and call back it in the limit function in planner.c like this:
![planner c](https://github.com/user-attachments/assets/bca4724d-dfe5-4b7c-a4ec-b04548864c7c)


and then set and reset the gc_state.modal.activeaccelprofile to whatever profile is currently needed with the g187 function?